### DISCUSSION TITLE
Merging Layers

### DISCUSSION THREAD

**Discussion by lookme2** [Argmaster/pygerber#102] [General]
I am working on merging the silk screen and top copper of a board.  Can this be done without saving two sperate image files?

**Comment by Argmaster** [comment]
It depends what do you mean by "without saving". 

If what you want is to avoid creation of two separate image files on drive, then yes, it is possible via API, with [`Rasterized2DResultHandle.get_image()`](https://argmaster.github.io/pygerber/v2.1.0/reference/pygerber/gerberx3/api/_layers.html#pygerber.gerberx3.api._layers.Rasterized2DResultHandle), then you should be able to operate on Image objects from Pillow library to [simply merge them programatically](https://www.tutorialspoint.com/python_pillow/Python_pillow_merging_images.htm), eg. by pasting with mask. I'm not sure but it might be necessary to use ColorScheme which include alpha channel and use RGBA images for this purpose.

However, if you were asking about rendering two images in batch (so you won't have to merge them manually) for example with command line, unfortunately it is not possible (yet).

In latter case may consider creating issue for it with description of what you need and how you want it to work.

**Comment by lookme2** [comment]
Thanks for the reply.  I think I'll give the Rasterized2DResultHandle a try.

**Comment by Argmaster** [comment]
@lookme2 I have just deployed PyGerber 2.3.0 which provides native way for doing exactly that:

```python
from pygerber.examples import ExamplesEnum, load_example
from pygerber.gerberx3.api.v2 import FileTypeEnum, GerberFile, Project

Project(
    [
        GerberFile.from_str(
            load_example(ExamplesEnum.simple_2layer_F_Cu),
            FileTypeEnum.COPPER,
        ),
        GerberFile.from_str(
            load_example(ExamplesEnum.simple_2layer_F_Mask),
            FileTypeEnum.MASK,
        ),
        GerberFile.from_str(
            load_example(ExamplesEnum.simple_2layer_F_Paste),
            FileTypeEnum.PASTE,
        ),
        GerberFile.from_str(
            load_example(ExamplesEnum.simple_2layer_F_Silkscreen),
            FileTypeEnum.SILK,
        ),
    ],
).parse().render_raster("output.png", dpmm=40)
```


<p align="center">
  <img width="400" src="https://github.com/Argmaster/pygerber/assets/56170852/9b3f3823-67b3-49f1-8c76-e2bddaca81fe">
</p>
